ext-apiserver/pkg/controller/machine/controller.go

```diff
@@ -39,13 +40,14 @@ type MachineControllerImpl struct {
 
 	// lister indexes properties about Machine
 	lister listers.MachineLister
-
 	// lister indexes properties about Cluster
 	clusterLister listers.ClusterLister
 
 	actuator cloud.MachineActuator
 
-	clientSet *clientset.Clientset
+	clientSet     *clientset.Clientset
+	machineClient v1alpha1.MachineInterface
+	linkedNodes   map[string]bool
```

Comment:
What is protecting this from concurrent access?

Reply:
+1

Reply:
It's only called from NodeReconcile which is only called from main queueing processing function. There should be no race.

Reply:
Could you add a comment to NodeReconcile to indicate it is not threadsafe. I would not want someone in the future assume that can safely do a performance boost by making it run in parallel for all nodes.